server/routes.go

```diff
@@ -445,3 +446,21 @@ func putVersionsHandler(m KeyManager, principal knox.Principal, parameters map[s
 		return nil, errF(knox.InternalServerErrorCode, err.Error())
 	}
 }
+
+func authorizeRequest(key *knox.Key, principal knox.Principal, access knox.AccessType) bool {
+	if principal.CanAccess(key.ACL, access) {
+		return true
+	}
+
+	input := knox.AccessCallbackInput{
+		Key:        *key,
+		Principals: principal.Raw(),
+		AccessType: access,
+	}
+
+	if accessCallback != nil {
+		return accessCallback(input)
```

Comment:
I think this could also panic. maybe we should recover + log panic, then return false. because who knows what the callback could be doing 
https://gobyexample.com/recover